letta/agent.py

```diff
@@ -844,8 +844,29 @@ def _handle_ai_response(
                 self.interface.function_message(f"Error: {error_msg}", msg_obj=messages[-1])
                 return messages, False, True  # force a heartbeat to allow agent to handle error
 
+            # Step 4: check if function response is an error
+            if function_response_string.startswith("Error"):
```

Comment:
where is this `"Error"` substring coming from? Could we maybe make this a constant (e.g. `SANDBOX_ERROR_MSG_PREFIX` so that we don't accidentally break this later on if we are modifying the function response elsewhere?

Reply:
sure I'll pull it out into a constant and use it where I'm generating the error message

> following up with a more robust implementation that doesn't infer status based on the string (but this is fine for now since I am setting the string the same way at all callsites)

mentioned in the summary but I'll make this more robust in my follow up (tomorrow)!